modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +209,158 @@ TEST(GPU, Symm7x7_test)
+    cl_mem       image_mem = nullptr;
+    cl_context   context = nullptr;
+    cl_device_id device = nullptr;
+
+    cv::directx::convertFromD3D11Texture2DtoCLMem(texture, &image_mem, &context, &device);
+
+    if (context == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_mem\n");
+    }
+    if (image_mem == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_context\n");
+    }
+    if (device == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLMem returns empty cl_device_id\n");
+    }
+
+    // Move data to host
+    std::vector<uint8_t> out_test_data(width * height, 0);
+    // Target data
+    const size_t sz0[3] = {0, 0, 0};
+    const size_t sz1[3] = {width, height, 1};
+
+    cl_command_queue q = (cl_command_queue)cv::ocl::Queue::getDefault().ptr();
+    cl_int status = clEnqueueReadImage(q, image_mem, CL_FALSE, sz0, sz1,
+                                       width * sizeof(uint8_t),
+                                       height * sizeof(uint8_t),
+                                       &out_test_data.front(), 0, NULL, NULL);
+    if (status != CL_SUCCESS) {
+        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: clEnqueueReadImage failed");
+    }
+    clFinish(q);
+    clReleaseMemObject(image_mem);
+    clReleaseContext(context);
+    clReleaseDevice(device);
+    // Comparison /////////////////////////////////////////////////////////////
+    EXPECT_EQ(out_test_data, test_data);
+}
+
+TEST(GPU_D3D11, ConvTexture2DtoUMat)
+{
+    // Create test data ///////////////////////////////////////////////////////
+    const UINT width = 150;
+    const UINT height = 100;
+    ID3D11Texture2D* texture = nullptr;
+    std::vector<uint8_t> test_data(width * height, 0);
+
+    prepareD3D11Texture2D(width, height, test_data, &texture);
+
+    // Convert texture2D to cl_mem_image //////////////////////////////////////
+    cl_mem       image_mem = nullptr;
+    cl_context   context = nullptr;
+    cl_device_id device = nullptr;
+    cv::directx::convertFromD3D11Texture2DtoCLMem(texture, &image_mem, &context, &device);
+
+    if (context == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLImage returns empty cl_mem\n");
+    }
+    if (image_mem == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLImage returns empty cl_context\n");
+    }
+    if (device == nullptr) {
+        CV_Error(cv::Error::BadDataPtr, "OpenCL: convertFromD3D11Texture2DtoCLImage returns empty cl_device_id\n");
+    }
+
+    // Convert to UMat function ///////////////////////////////////////////////
+    // Create cv::Umat
+    cv::UMat umt_test(USAGE_ALLOCATE_DEVICE_MEMORY);
+    // Copy data from cl_mem_image to cv::Umat buffer
+    cv::ocl::convertFromImage(image_mem, umt_test);
```

Comment:
is there any ownership constructor from handle without copy?

Reply:
Buffer -> Image.. and Image.. -> Buffer can't avoid copy. If we get D3D11Buffer, I think, we will use convertFromBuffer instead that get ownership